src/SetupApi.Desktop/SetupApi.cs

```diff
@@ -0,0 +1,195 @@
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class SetupApi
+    {
+        /// <summary>
+        /// The SetupDiGetClassDevs function returns a <see cref="SafeDeviceInfoSetHandle" /> handle to a device information set
+        /// that contains requested device information elements for a local computer.
+        /// </summary>
+        /// <param name="classGuid">
+        /// A pointer to the GUID for a device setup class or a device interface class. This pointer is
+        /// optional and can be <see langword="null" />. For more information about how to set ClassGuid, see the following Remarks
+        /// section.
+        /// </param>
+        /// <param name="enumerator">
+        /// A pointer to a NULL-terminated string that specifies:
+        /// <list type="bullet">
+        ///     <item>
+        ///         <description>
+        ///         An identifier(ID) of a Plug and Play(PnP) enumerator.This ID can either be the value's globally
+        ///         unique identifier (GUID) or symbolic name. For example, "PCI" can be used to specify the PCI PnP value. Other
+        ///         examples of symbolic names for PnP values include "USB," "PCMCIA," and "SCSI".
+        ///         </description>
+        ///     </item>
+        ///     <item>
+        ///         <description>
+        ///         A PnP device instance ID.When specifying a PnP device instance ID, DIGCF_DEVICEINTERFACE must be
+        ///         set in the Flags parameter.
+        ///         </description>
+        ///     </item>
+        /// </list>
+        /// This pointer is optional and can be <see langword="null" />.If an enumeration value is not used to select devices, set
+        /// Enumerator to <see langword="null" />.
+        /// </param>
+        /// <param name="hwndParent">
+        /// A handle to the top-level window to be used for a user interface that is associated with
+        /// installing a device instance in the device information set. This handle is optional and can be <see langword="null" />.
+        /// </param>
+        /// <param name="flags">
+        /// A variable of type DWORD that specifies control options that filter the device information elements
+        /// that are added to the device information set. This parameter can be a bitwise OR of zero or more of the flags.
+        /// </param>
+        /// <returns>
+        /// If the operation succeeds, SetupDiGetClassDevs returns a handle to a device information set that contains all
+        /// installed devices that matched the supplied parameters. If the operation fails, the function returns an invalid handle.
+        /// </returns>
+        [DllImport(nameof(SetupApi), SetLastError = true)]
+        public static extern SafeDeviceInfoSetHandle SetupDiGetClassDevs(
+            IntPtr classGuid,
+            [MarshalAs(UnmanagedType.LPStr)] string enumerator,
```

Comment:
I understand that `LPTStr` is the default marshaling for string. I'm guessing `LPStr` means always use ANSI instead of Unicode. Would this attribute then be equivalent to simply adding `CharSet = CharSet.Ansi` to the `DllImport` attribute? I'm thinking declaring it at the method level rather than the parameter level would be easier to read.